dev-scheme/guile/guile1-1.8.8.recipe

```diff
@@ -0,0 +1,146 @@
+
+commandSuffix=$secondaryArchSuffix
+commandBinDir=$binDir
+if [ "$targetArchitecture" = x86_gcc2 ]; then
+	commandSuffix=
+	commandBinDir=$prefix/bin
+fi
+
+libguileVersion="17.4.0"
+libguileVers="${libguileVersion%.*}"
+libguileVersionCompat="$libguileVersion compat >= ${libguileVersion%%.*}"
+portVersionCompat="$portVersion compat >= ${portVersion%%.*}"
+portVers="${portVersion%.*}"
+
+PROVIDES="
+	guile1$secondaryArchSuffix = $portVersionCompat
+	cmd:guile1$commandSuffix = $portVersion
+	cmd:guile1_snarf$commandSuffix = $portVersion
+	cmd:guile1_tools$commandSuffix = $portVersion
+	lib:libguile$secondaryArchSuffix = $libguileVersionCompat
+	lib:libguilereadline_v_17$secondaryArchSuffix = $libguileVersionCompat
+	lib:libguile_srfi_srfi_1_v_3$secondaryArchSuffix = 3.0.2
+	lib:libguile_srfi_srfi_13_14_v_3$secondaryArchSuffix = 3.0.1
+	lib:libguile_srfi_srfi_4_v_3$secondaryArchSuffix = 3.0.1
+	lib:libguile_srfi_srfi_60_v_2$secondaryArchSuffix = 2.0.2
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libffi$secondaryArchSuffix
+	lib:libgc$secondaryArchSuffix
+	lib:libgmp$secondaryArchSuffix
+	lib:libiconv$secondaryArchSuffix
+	lib:libintl$secondaryArchSuffix
+	lib:libltdl$secondaryArchSuffix
+	lib:libncurses$secondaryArchSuffix
+	lib:libreadline$secondaryArchSuffix
+	lib:libunistring$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	guile1${secondaryArchSuffix}_devel = $portVersionCompat
+	cmd:guile$commandSuffix = $portVersion
+	cmd:guile_config$commandSuffix = $portVersion
+	devel:libguile$secondaryArchSuffix = $libguileVersionCompat
+	devel:libguilereadline_v_17$secondaryArchSuffix = $libguileVersionCompat
+	devel:libguile_srfi_srfi_1_v_3$secondaryArchSuffix = 3.0.2
+	devel:libguile_srfi_srfi_13_14_v_3$secondaryArchSuffix = 3.0.1
+	devel:libguile_srfi_srfi_4_v_3$secondaryArchSuffix = 3.0.1
+	devel:libguile_srfi_srfi_60_v_2$secondaryArchSuffix = 2.0.2
+	"
+REQUIRES_devel="
+	guile1$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libffi$secondaryArchSuffix
+	devel:libgc$secondaryArchSuffix
+	devel:libgmp$secondaryArchSuffix
+	devel:libiconv$secondaryArchSuffix
+	devel:libintl$secondaryArchSuffix
+	devel:libltdl$secondaryArchSuffix
+	devel:libncurses$secondaryArchSuffix
+	devel:libreadline$secondaryArchSuffix
+	devel:libunistring$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:awk
+	cmd:gcc$secondaryArchSuffix
+	cmd:gperf
+	cmd:grep
+	cmd:make
+	cmd:ld
+	cmd:pkg_config$secondaryArchSuffix
+	cmd:sed
+	"
+
+defineDebugInfoPackage guile1$secondaryArchSuffix \
+	"$libDir"/libguile.so.$libguileVersion
+
+BUILD()
+{
+	CPPFLAGS="-D_BSD_SOURCE" LIBS="-lbsd -lnetwork" \
+	runConfigure --omit-dirs binDir ./configure \
+		--bindir=$commandBinDir \
+		--with-threads --disable-error-on-warning
+
+	# No MADV_DONTNEED, yet
+	sed --in-place '/HAVE_SYS_MMAN_H/d' config.h
+
+	make $jobArgs
+}
+
+INSTALL()
+{
+	make install
+
+# rename to avoid conflicts with newer guile verions (allow parallel install)
```

Comment:
Split the binaries into a subpackage instead, please. Change the 2.x recipe accordingly.